javascript/ql/src/semmle/javascript/frameworks/Electron.qll

```diff
@@ -33,4 +33,38 @@ module Electron {
       this = DataFlow::moduleMember("electron", "BrowserView").getAnInstantiation()
     }
   }
+  
+  /**
+   * A Node.js-style HTTP or HTTPS request made using `electron.net`.
+   */
+  class NetRequest extends NodeJSLib::ClientRequest {
+    NetRequest() {
+      exists(DataFlow::MethodCallNode mcn |
+        mcn = this and
+        DataFlow::moduleMember("electron", "net").flowsTo(mcn.getReceiver()) and
+        mcn.getMethodName() = "request"
+      )
+    }
+    
+    override DataFlow::Node getOptions() {
+      result = this.(DataFlow::MethodCallNode).getArgument(0)
+    }
+  }
+  
+  
+  /**
+   * A Node.js-style HTTP or HTTPS request made using `new Electron.client`.
+   */
+  class NewClientRequest extends NodeJSLib::ClientRequest {
+    NewClientRequest() {
+      exists(DataFlow::NewNode nn |
+        this = nn and
+        DataFlow::moduleMember("electron", "ClientRequest").flowsTo(nn.getCallee())
```

Comment:
`this = DataFlow::moduleMember("electron", "ClientRequest").getAnInstantiation()`